docs/source/basic/terms.rst

```diff
@@ -0,0 +1,65 @@
+Terms
+=====
+
+This page provides an overview of the terms used in ``alpaka`` and the relationships between them.
+
+.. image:: /images/alpaka_terms.svg
+
+Platform
+--------
+
+- A ``platform`` contains information about the system, e.g. the available devices. 
+- Depending on the platform, it also contains a runtime context.
+- A ``platform`` has a least one device but it can also has many device.
+- Each ``platform`` can be used with any number of ``accelerator``. ``platforms`` and ``accelerator`` cannot freely combined. An ``accelerator`` supports only a specific ``platform``.
+ 
+Device
+------
+
+- A ``device`` represent a compute unit, such as a CPU or a GPU.
+- Each ``device`` is bounded to a specific ``platform``.
+- Each ``device`` can have any number of ``queues``.
+
+Accelerator
+-----------
+
+- A ``accelerator`` is a index mapping function. It distributes the index space to the chunks. The ``accelerator`` maps a continues index space to a blocked index domain decomposition.
+- It is not allowed to create an instance of an ``accelerator``.
+- A ``accelerator`` is bounded to a specific ``platform``.
```

Comment:
This could be important. "Accelerator makes the kernel device-agnostic". Accelerator maps device side functions and provides access to shared memory ?